api/envoy/extensions/filters/http/grpc_json_transcoder/v3/transcoder.proto

```diff
@@ -270,4 +270,12 @@ message GrpcJsonTranscoder {
   // Proto enum values are supposed to be in upper cases when used in JSON.
   // Set this to true if your JSON request uses non uppercase enum values.
   bool case_insensitive_enum_parsing = 14;
+
+  // The maximum size of a body to be transcoded, in bytes. A body exceeding this size will
+  // provoke a ``HTTP 413 Request Entity Too Large`` if the request body was oversized,
+  // or ``HTTP 500 Internal Server Error`` response with a descriptive message if the response
+  // body was oversized.
+  //
+  // If unset or zero, the current stream buffer size is used.
```

Comment:
I think a primitive type cannot be unset, and if this is intended, then you can convert it to a WKT (`google.protobuf.UInt32Value`).
Also should there be a maximal value? (maybe not, but want to ensure that this is correct).

CC'ing @KBaichoo and @nareddyt to review from a filter's point-of-view and ensuring that the default behavior is correct.

Reply:
Unsure whether to WKT it to be explicit that unset is a special case (and then constrain it so zero is a config error), or just do what I've done; I've removed the word "unset" for now so it's unambiguous.

Reply:
I don't think there's any maximum - grpc itself has a soft constraint at, I think, 4 or 6MB per message, but that constraint can be overridden, so I don't think there's any justification for a hard limit here. (There's no hard limit on setDecoderBufferLimit, which when the value is high this is essentially a proxy for.)

Reply:
Among other changes, I have now changed it to WKT with explicitly forbidden zero, because it makes some of the code more readable to use `absl::optional` and then `value_or` rather than using zero as a sentinel value. (Though it makes the tests more of a pain!)